tests/sesolve/closed_system.py

```diff
@@ -0,0 +1,64 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+import qutip as qt
+import torch
+from torch import Tensor
+
+import torchqdynamics as tq
+
+
+class ClosedSystem(ABC):
+    def __init__(self):
+        self.n = None
+        self.H = None
+        self.H_batched = None
+        self.psi0 = None
+        self.psi0_batched = None
+        self.exp_ops = None
+
+    @abstractmethod
+    def t_save(self, n: int) -> Tensor:
+        pass
+
+    def psi(self, t: float) -> Tensor:
```

Comment:
If it were then it would be mandatory to define this method (one can't instantiate an abstract class), but I thought we may have some physical example with no analytical solutions, for which we don't define this guy.

Reply:
But then it could be the solution given by qutip for instance, no? In case we don't have an analytical solution.

Reply:
Yes, but I prefer not to make it mandatory, I think we may want to test specific things on some systems without needing the full state (expectation values for example, or just that the solver is stable), so we should not make it mandatory to have a solution for `psi` or `rho`.